qiskit/providers/aer/backends/aerbackend.py

```diff
@@ -288,9 +288,30 @@ def _run(self, qobj, job_id='', format_result=True):
         # Start timer
         start = time.time()
 
+        # Take metadata from headers of experiments to work around JSON serialization error
+        metadata_list = []
+        metadata_index = 0
+        for expr in qobj.experiments:
+            if hasattr(expr.header, "metadata"):
+                metadata_copy = expr.header.metadata.copy()
+                metadata_list.append(metadata_copy)
+                expr.header.metadata.clear()
+                if "id" in metadata_copy:
+                    expr.header.metadata["id"] = metadata_copy["id"]
```

Comment:
What is this `id` for/from? and why are we preserving it in the qobj?

Reply:
This is my mistake. I should reduce these codes.

Qiskit-Aer internally use this metadata for DASK execution. Even though metadata is cleared, this `metadata['id']` was necessary for merging results and to be kept. However, the latest commit of this PR keeps whole metadata, then we do not need to specialize it.

I reduced the codes in 461077b. Thanks.